grants/speculative/Dkg_Multisig_Wallet.md

```diff
@@ -0,0 +1,152 @@
+   * Adapter in Rust (200h)
+      * Write all the methods for external calls - 168h
+         * NewDistKeyGen
+         * ProcessDeal
+         * ProcessResponse
+         * ProcessSecretCommits
+         * DistKeyShare
+         * NewDSS
+         * ProcessPartialSignature
+         * ConstructFinalSignature
+         * VerifySignature
+      * Integrate adapter with DKG library - 8h
+      * Write unit tests for Rust code - 16h
+      * Write unit tests for new compiled JS code - 8h
+   * Integration testing adapter with application (16h)
+   * Integration testing application with server (8h)
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Total for Wallet App: 384h - 1 back-end and 1 front-end developers (2 months)
+
+2. Web server
+   * API written with actix
+      * Write all the endpoints with asynchronous functionality - 124h
+         * NewWallet
+         * JoinWallet
+         * BroadcastDeal
+         * CreateNewTransaction
+         * BroadcastPartialSignature
+         * BroadcastFinalSignature
+      * Write functionality for authentication and encryption - 16h
+         * NewUser
+         * CheckUser
+         * Encrypt
+         * Decrypt
+      * Integrate functionality with DB - 4h
+      * Write unit tests - 16h
+      * Configure all this with Docker to easier deploy it to server - 8h
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Total for Web server: 168h - 1 back-end developer (1 month)
+
+3. DKG library
+   * VSS (240h)
+      * Implement polynomial commitments with Lagrange interpolation - 72h
+         * NewPrivPoly
+         * RecoverSecret
+         * NewPubPoly
+      * Implement Dealer functionality (struct which will create shares and deals(set of data such polynomials, sessionID, threshold)) - 56h
+         * NewDealer
+         * EncryptDeal
+         * SecretCommit
+      * Implement Verifier functionality (struct which will receive deals from all the participants and verify it) - 96h
+         * NewVerifier
+         * ProcessEncryptedDeal
+      * Write unit tests - 16h
+   * DKG(320h)
+      * Integrate with VSS functionality - 16h
+      * Write functionality to create and process commitments of the secret polynomials - 56h
+         * NewDistKeyGenerator
+         * ProcessDeals
+         * ProcessSecretCommits
+      * Write reconstruction key functionality - 32h
+         * GenDistKeyShare
+      * Write unit tests - 16h
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Total for DKG library: 560h - 2 back-end developers (2.5 months)
+
+4. DSS library
+   * Integrate with DKG functionality - 8h
+   * Write functionality to produce and check partial signature - 62h
+      * ProducePartialSig
+      * ProcessPartialSig
+   * Write functionality to compute final distributed signature from partial signatures - 56h
+      * ComputeFinalSignature
+   * Write unit tests - 16h
+   * Write integration tests with DKG - 16h
```

Comment:
We should make signature generation part of schnorrkel because some will want distributed private keys doing other stuff that schnorrkel keys do.